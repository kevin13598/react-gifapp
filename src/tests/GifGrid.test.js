import {shallow} from 'enzyme'
import { GifGrid } from '../components/GifGrid';
import '@testing-library/jest-dom';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs');

describe('Pruebas de <GifGrid/>', () =>{
    
    const category = 'One Punch Man';

    test('Debe de mostrar el componente correctamente.', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        })

        const wrapper = shallow(<GifGrid category = {category}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [
        {
            id: 'abc',
            url: 'https://localhost/cualquiercosa/fd.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: 'abcd',
            url: 'https://localhost/cualquiercosa/fd.jpg',
            title: 'Cualquier cosa 2'
        }
    ];
        
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category = {category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


    })
    

});