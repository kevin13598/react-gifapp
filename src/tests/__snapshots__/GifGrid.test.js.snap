// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid/> Debe de mostrar el componente correctamente. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
    One Punch Man
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas de <GifGrid/> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
    One Punch Man
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa/fd.jpg"
    />
    <GifGridItem
      id="abcd"
      key="abcd"
      title="Cualquier cosa 2"
      url="https://localhost/cualquiercosa/fd.jpg"
    />
  </div>
</Fragment>
`;
